backend/typescript/services/implementations/campService.ts

```diff
@@ -102,10 +102,18 @@ class CampService implements ICampService {
   }
 
   async updateCampById(campId: string, camp: UpdateCampDTO): Promise<CampDTO> {
-    let oldCamp: Camp | null;
+    const oldCamp: Camp | null = await MgCamp.findById(campId);
+
+    if (!oldCamp) {
+      throw new Error(`Camp' with campId ${campId} not found.`);
+    }
+
+    if (oldCamp.active && camp.fee) {
+      throw new Error(`Error - cannot update fee of active camp`);
+    }
```

Comment:
I think we still want these errors to be contained within their own try catch so we can still throw `Logger.error` which is used for debugging in the console:
```
let oldCamp: Camp | null;
try{
    oldCamp = await MgCamp.findById(campId);

    if (!oldCamp) {
      throw new Error(`Camp' with campId ${campId} not found.`);
    }

    if (oldCamp.active && camp.fee) {
      throw new Error(`Error - cannot update fee of active camp`);
    }
}
catch (error: unknown) {
      Logger.error(`Failed to update camp. Reason = ${getErrorMessage(error)}`);
      throw error;
}
```